src/python/bcc/__init__.py

```diff
@@ -388,7 +388,6 @@ def attach_kprobe(self, event="", fn_name="", event_re="",
         return self
 
     def detach_kprobe(self, event):
-        assert isinstance(event, str), "event must be a string"
```

Comment:
If it's not being hit for some existing tests or obvious use cases, I'd say leave the check then. Or is there some other concern?